### ISSUE TITLE
Memory leak in monai.transforms.distance_transform_edt

### ISSUE THREAD

**Issue by aaronkujawa** [Project-MONAI/MONAI#7727]
**Describe the bug**
There seems to be a memory leak in monai.transforms.distance_transform_edt when cuCIM is not available. 

**To Reproduce**
cuCIM must not be available to reproduce
```
import torch
from monai.transforms import distance_transform_edt

nb_interations = 2000
for i in range(nb_interations):
    mask = torch.randint(low=0, high=2, size=(10, 10, 10,), device='cuda')
    distance_transform_edt(mask)

print(torch.cuda.memory_summary(device=None, abbreviated=True))
```

**Expected behavior**
The GPU memory footprint increases with nb_iterations. This should not happen. This does not happen when cuCIM is installed.

**Additional context**
The problem seems to be an [exception leak](https://cosmicpercolator.com/2016/01/13/exception-leaks-in-python-2-and-3/) in optional_import:

https://github.com/Project-MONAI/MONAI/blob/56508992c32e328667740e18f99b82a6a19f2c2a/monai/utils/module.py#L398-L408

where tb does not get cleared. tb holds references that prevent the mask variable from being deleted. I think this could be a problem with other optional_import calls, too.

**Comment by johnzielke** [comment]
Most likely related to: https://github.com/Project-MONAI/MONAI/issues/7480
